File: pilha/Pilha Estática/Java/Pilha Invertida/Empilhavel.java
public interface Empilhavel {
	//métodos principais
	void empilhar(Object dado);	//C
	Object espiar();			//R
	void atualizar(Object dado);//U
	Object desempilhar();		//D
	
	//métodos auxiliares
	boolean estaCheia();
	boolean estaVazia();
	String imprimir();
}

File: pilha/Pilha Estática/Kotlin/Pilha Genérica/Empilhavel.kt
interface Empilhavel<T> {
    // métodos principais
    fun empilhar(dado: T?)
    fun desempilhar(): T?
    fun espiar(): T?
    fun atualizar(dado: T?)      
    // métodos auxiliares
    fun estaCheia(): Boolean
    fun estaVazia(): Boolean
    fun imprimir(): String
}
